app/lib/actions/katello/host/register.rb

```diff
@@ -29,7 +29,6 @@ def plan(host, system, consumer_params, content_view_environment, activation_key
             host.save!
 
             action_subject host
-            connect_to_smart_proxy(host)
```

Comment:
At this point, the `host` object does not have an organization directly? Does `host.content_host` have one? The `host.content_host` object is assigned a system which has a lifecycle_environment which has an organization. That is a round about way of figuring all of that out but I was curious. I think your change is generally fine, I am just less familiar with run vs finalize.